src/kaiku.ts

```diff
@@ -1446,6 +1446,61 @@ function h(
   })
 }
 
+function jsx(
+  type: HtmlElementTagName,
+  props: HtmlElementProperties,
+  key?: string
+): HtmlElementDescriptor
+function jsx<PropertiesT extends DefaultProps>(
+  type: FunctionComponent<PropertiesT>,
+  props: PropertiesT | null,
+  key?: string
+): FunctionComponentDescriptor<PropertiesT>
+function jsx<PropertiesT extends DefaultProps>(
+  type: ClassComponent<PropertiesT>,
+  props: PropertiesT | null,
+  key?: string
+): ClassComponentDescriptor<PropertiesT>
+function jsx(
+  type: typeof Fragment,
+  props: null | { key?: string | number },
+  key?: string
+): FragmentDescriptor
+function jsx(
+  type: any,
+  props: DefaultProps | HtmlElementProperties | null,
+  key?: string
+): NodeDescriptor<any> {
+  const propsCopy: Record<string, any> = {
+    ...props,
+    key,
+  }
+  let children = propsCopy.children
+  children = children !== undefined ?
+    (Array.isArray(children) ? children : [children]) : undefined
+  propsCopy.children = children
+
+  if (typeof type === 'string') {
+    delete propsCopy.children
```

Comment:
instead of doing `delete` statemens for each case, maybe do the assignment of `children` instead